tests/test_django_settings_module.py

```diff
@@ -359,6 +359,41 @@ def test_cfg(pytestconfig):
     assert r.ret == 0
 
 
+def test_no_ds_but_django_conf_imported(testdir, monkeypatch):
+    """pytest-django should not bail out, if "django.conf" has been imported
+    somewhere, e.g. via hypothesis (#599)."""
+
+    monkeypatch.delenv('DJANGO_SETTINGS_MODULE')
+
+    testdir.makepyfile("""
+        import os
+        import sys
+
+        # line copied from hypothesis/extras/django.py
+        from django.conf import settings as django_settings
+
+        # Don't let pytest poke into this object, generating a
+        # django.core.exceptions.ImproperlyConfigured
+        del django_settings
```

Comment:
Wow.. `inspect.isclass` / `isinstance` triggers the ImproperlyConfigured already..